### DISCUSSION TITLE
Automatic Git push

### DISCUSSION THREAD

**Discussion by eliasu** [statamic/cms#7211] [Q&A]
Hi all, 

I setup [Git Integration](https://statamic.dev/git-automation) for my project and it works. But I have to `git push` manually via SSH. 

After some reading about Queue Drivers, I realized that `QUEUE_CONNECTION=sync` won't work for the delay und auto-push functionality. So I thought about ´Redis´ or something else. 

I' not using Ploi, Force, etc but shared hosting. SSH is available, but some commands are blocked. For example I'm not able to install Redis, or at least I don't get it how (`sudo` or `apt-get` are unknown commands). 

So do you have any ideas, how to accomplish automatic pushes (delaying commits, is not that important)? Or should I use a Cronjob for this?

Thanks for any advice!

**Comment by duncanmcclean** [comment]
I'm guessing you've already [enabled pushing](https://statamic.dev/git-automation#pushing-changes) when configuring the Git integration?

![image](https://user-images.githubusercontent.com/19637309/207313922-82b35f3f-f5c9-4588-bde4-3f83373a67c7.png)

**Reply by eliasu** [reply]
Sure. But it does not work, since it needs a Queue Worker like Redis or something. Or am I mistaken?

**Reply by duncanmcclean** [reply]
It doesn't *need* a queue worker - it'll work without one.

**Reply by duncanmcclean** [reply]
Is there anything in your logs?

**Reply by eliasu** [reply]
Oh no, I should have checked … you're right. Log says: 
```
LOG.error: Git Process: Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Ok, got the problem then 👍 
But can you explain, why am I able to push manually via SSH? I assumed if that is possible, the setup is correct?

**Reply by eliasu** [reply]
I can't get it done!
Could you provide me steps to solve this?

Btw: manually pushing is still possible.

**Reply by duncanmcclean** [reply]
Just looking at previous issues where this error message has come up.... Could [this](https://github.com/statamic/cms/issues/2191#issuecomment-671099989) be what's happening?
